repm/api.go

```diff
@@ -176,6 +213,73 @@ func (conn *connection) dispatchPullProgress(reqBytes []byte) ([]byte, error) {
 	return mustMarshal(res), nil
 }
 
+func (conn *connection) newTransaction() int {
+	conn.transactionMutex.Lock()
+	defer conn.transactionMutex.Unlock()
+	txID := conn.transactionCounter
+	conn.transactionCounter++
+	conn.transactions[txID] = conn.db.NewTransaction()
+	return txID
+}
+
+func (conn *connection) dispatchBeginTransaction(reqBytes []byte) ([]byte, error) {
+	var req beginTransactionRequest
+	err := json.Unmarshal(reqBytes, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	txID := conn.newTransaction()
+
+	res := beginTransactionResponse{
+		TransactionID: txID,
+	}
+	return mustMarshal(res), nil
+}
+
+func (conn *connection) dispatchCloseTransaction(reqBytes []byte) ([]byte, error) {
```

Comment:
ocd brain nit: begin/close mismatch, begin/end or open/close?

Reply:
Agreed. Switch to open/close